File: 프로그래머스/lv2/42747. H－Index/H－Index.js
function solution(citations) {
    let hIndex = 0;
    citations.sort((a, b) => b - a);
    for (let i = 0; i < citations.length; i++) {
        if (citations[i] >= i + 1) {
            hIndex = i + 1;
            console.log(citations[i], '>=', i, '=', citations[i] >= i)
        }
    }
    return hIndex;
}

File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    h_index = 0 # H-Index : 인용 횟수가 h 이상인 논문이 h편 이상일 때의 최댓값
    citations.sort(reverse = True)
    
    for i in range(len(citations)):
        if citations[i] >= i + 1:
            h_index += 1
    return h_index

